src/lib/logstore/log_store_service.cpp

```diff
@@ -175,7 +175,9 @@ void LogStoreService::open_logdev(logdev_id_t logdev_id) {
         m_id_reserver->reserve(logdev_id);
         auto logdev = std::make_shared< LogDev >(logdev_id, m_logdev_vdev.get());
         m_id_logdev_map.emplace(logdev_id, logdev);
+        LOGDEBUGMOD(logstore, "log_dev={} does not exist, created!", logdev_id);
```

Comment:
We already clear it in line 160

Reply:
yes, but that is the moment we start logservice, right? 
we open the logdev before we start logservice, so according to the logic here, it will crerate a new one, right?

Reply:
I really dont get the issue you are tyring to fix. Wont the changes in home_raft_log_store.cpp fix the actual issue ?

Reply:
no, the changes in home_raft_log_store.cpp is to prevent the repl_dev from joining raft group before the repl_log_store is ready. this is one changes.

the direct cause of the recovery issue is we do not remove open logdevs from m_unopened_logdev correctly. other changes in the PR is to resolve this.